Feat/feature screenone data module

anarahari opened the pull request.

anarahari committed:
feat : Feature screen one, domain module setup

anarahari committed:
Merge branch 'feat/Workflow_EmptyModules' into feat/Feature_Screen_One_Domain

# Conflicts:
#	common/build.gradle.kts

anarahari committed:
feat : use case, repository, pagination setup

anarahari committed:
Merge branch 'feat/Workflow_EmptyModules' into feat/Feature_Screen_One_Domain

anarahari committed:
feat : setup data layer for screen one

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

# Conflicts:
#	domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt
#	domain/src/main/java/com/compose/domain/repository/CharacterRepository.kt
#	domain/src/main/java/com/compose/domain/usecase/GetCharactersUseCase.kt

anarahari committed:
feat : screen one data module setup with graphql, repo impl, dagger modules

shreyansh-mm commented on common/src/main/java/com/compose/common/CommonConstants.kt:
```diff
@@ -0,0 +1,5 @@
+package com.compose.common
+
+object CommonConstants {
```
Should be marked as internal

anarahari replied:
done

shreyansh-mm commented on data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt:
```diff
@@ -0,0 +1,23 @@
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class CharacterRepositoryImpl @Inject constructor(private val apolloClient: ApolloClient) : CharacterRepository {
```
Let's see if we can make this class `internal`

anarahari replied:
ok let me try

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : Added datamapper, repositoryimpl changes to integrate data mapper for character model

anarahari committed:
feat : ktlintcheck changes for spaces

anarahari committed:
feat : CommonConstants marked as internal

shreyansh-mm commented on data/src/main/java/com/compose/data/datamapper/DataMapper.kt:
```diff
@@ -0,0 +1,15 @@
+
+fun getCharactersQueryToCharacterModel(results: GetCharactersQuery.Result?): CharacterModel {
+    return CharacterModel(
+        id = results?.character?.id.orEmpty(),
+        name = results?.character?.name.orEmpty(),
+        image = results?.character?.image.orEmpty(),
+        gender = results?.character?.gender.orEmpty(),
+        species = results?.character?.species.orEmpty(),
+        status = results?.character?.status.orEmpty()
```
why do we have or empty?

anarahari replied:
orEmpty method checks if the string value is null, if the value is null, an empty string is returned

shreyansh-mm commented on data/src/main/java/com/compose/data/paging/CharacterPagingSource.kt:
```diff
@@ -17,19 +16,19 @@ class CharacterPagingSource(
+            // val response = apolloClient.query(GetCharactersQuery(Optional.present(page))).execute()
+            val response = apolloClient.query(GetCharactersQuery()).execute()
             // Check for errors in the response
             if (response.hasErrors()) {
                 LoadResult.Error(Exception(response.errors?.joinToString(", ") { it.message }))
             } else {
                 val characters = response.data?.characters?.results?.filterNotNull() ?: emptyList()
-                val nextKey =
-                    if (response.data?.characters?.info?.next != null) {
+                val nextKey = page + 1
+                    /*if (response.data?.characters?.info?.next != null) {
                         page + 1
                     } else {
                         null
-                    }
+                    }*/
```
This is being confusing to read, lets not keep code commented, remove them so it becomes easier to read.

shreyansh-mm commented on data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt:
```diff
@@ -0,0 +1,28 @@
+        val response = apolloClient.query(GetCharactersQuery()).execute()
+        if (response.hasErrors()) {
+            emit(emptyList())
+            return@flow
```
I don't think we need to use this, we can update the conditions above and make our code cleaner

anarahari committed:
feat : data

- CharacterPagingSource removed commented code

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : data

- Changed the implementation type of getCharacters to   List<T> in repository
- Nullable return of charactermodel in datamapper
